apps/prairielearn/src/ee/lib/ai-grading/ai-grading-stats.ts

```diff
@@ -191,7 +191,7 @@ export async function calculateAiGradingStats(
  */
 export async function selectGradingJobsInfo<T extends { id: string }>(
   instance_questions: T[],
-): Promise<Record<string, GradingJobInfo[]>> {
+): Promise<Partial<Record<string, GradingJobInfo[]>>> {
```

Comment:
I see where you're going with this, but I don't love this as a general pattern. It has the annoying side effect of things like `Object.values(...)` yielding an array with undefined values, and also means that one can deliberately write `undefined` into it.

See my other comments about the `... in ...` checks, I think this is related.

Reply:
What do you want to do here? A `Map`?

Reply:
A `Map` is probably better, yeah, though see my note about this in https://github.com/PrairieLearn/PrairieLearn/pull/12818#discussion_r2369608466.

Reply:
I undid this with my latest changes.